CI: ci-runtime: hive/job: TestTimer_ExitOnCloseFnCtx times out

## CI failure

```
panic: test timed out after 10m0s
running tests:
	TestTimer_ExitOnCloseFnCtx (10m0s)

goroutine 110 [running]:
testing.(*M).startAlarm.func1()
	/root/sdk/go1.22.0/src/testing/testing.go:2366 +0x385
created by time.goFunc
	/root/sdk/go1.22.0/src/time/sleep.go:177 +0x2d

goroutine 1 [chan receive, 9 minutes]:
testing.tRunner.func1()
	/root/sdk/go1.22.0/src/testing/testing.go:1650 +0x4ab
testing.tRunner(0xc0002eb380, 0xc00029fb58)
	/root/sdk/go1.22.0/src/testing/testing.go:1695 +0x134
testing.runTests(0xc000010fd8, {0x1273000, 0x14, 0x14}, {0x416354?, 0x7fe1de59d9d8?, 0x127ee00?})
	/root/sdk/go1.22.0/src/testing/testing.go:2159 +0x445
testing.(*M).Run(0xc000379cc0)
	/root/sdk/go1.22.0/src/testing/testing.go:2027 +0x68b
go.uber.org/goleak.VerifyTestMain({0xd604e0?, 0xc000379cc0?}, {0xc0000cde78, 0x1, 0x1})
	/host/vendor/go.uber.org/goleak/testmain.go:53 +0x5a
github.com/cilium/cilium/pkg/hive/job.TestMain(0xc000379cc0)
	/host/pkg/hive/job/job_test.go:36 +0x69
main.main()
	_testmain.go:121 +0x1b5

goroutine 115 [select]:
github.com/cilium/cilium/pkg/hive/cell.(*scope).start.func1()
	/host/pkg/hive/cell/structured.go:168 +0x10a
created by github.com/cilium/cilium/pkg/hive/cell.(*scope).start in goroutine 35
	/host/pkg/hive/cell/structured.go:164 +0x90

goroutine 35 [chan receive (nil chan), 9 minutes]:
github.com/cilium/cilium/pkg/hive/job.TestTimer_ExitOnCloseFnCtx(0xc00041ab60)
	/host/pkg/hive/job/job_test.go:622 +0x125
testing.tRunner(0xc00041ab60, 0xc9f3c8)
	/root/sdk/go1.22.0/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	/root/sdk/go1.22.0/src/testing/testing.go:1742 +0x390

goroutine 28 [syscall, 9 minutes]:
os/signal.signal_recv()
	/root/sdk/go1.22.0/src/runtime/sigqueue.go:152 +0x29
os/signal.loop()
	/root/sdk/go1.22.0/src/os/signal/signal_unix.go:23 +0x13
created by os/signal.Notify.func1.1 in goroutine 14
	/root/sdk/go1.22.0/src/os/signal/signal.go:151 +0x1f

goroutine 116 [chan receive, 9 minutes]:
github.com/cilium/cilium/pkg/hive/job.TestTimer_ExitOnCloseFnCtx.func1.1({0xd68290?, 0xc000729f50?})
	/host/pkg/hive/job/job_test.go:610 +0x89
github.com/cilium/cilium/pkg/hive/job.(*jobTimer).start(0xc00070d140, {0xd68290, 0xc000729f50}, 0x0?, {0xd681d0, 0xc00051f4a0}, {{{0x0, 0x0, 0x0}}, {0xd76ab0, ...}, ...})
	/host/pkg/hive/job/job.go:488 +0x511
created by github.com/cilium/cilium/pkg/hive/job.(*group).Start.func1 in goroutine 35
	/host/pkg/hive/job/job.go:162 +0x177
FAIL	github.com/cilium/cilium/pkg/hive/job	600.062s
```

--- a/pkg/hive/job/job_test.go
+++ b/pkg/hive/job/job_test.go
@@ -7,6 +7,7 @@ import (
 	"context"
 	"errors"
 	"runtime"
+	"sync"
 	"testing"
 	"time"
 
@@ -600,13 +601,11 @@ func TestTimer_ExitOnCloseFnCtx(t *testing.T) {
 	h := fixture(func(r Registry, s cell.Scope, l cell.Lifecycle) {
 		g := r.NewGroup(s)
 
+		var closer sync.Once
 		g.Add(
 			Timer("on-interval", func(ctx context.Context) error {
 				i++
-				if started != nil {
-					close(started)
-					started = nil
-				}
+				closer.Do(func() { close(started) })
 				<-ctx.Done()
 				return nil
 			}, 1*time.Millisecond),

